Do not install the vendored libshared static library and headers

Starting with 1.10.0, timewarrior uses

    add_subdirectory(${CMAKE_SOURCE_DIR}/src/libshared)

in src/CmakeLists.txt which inherits libshared's installation rules and
installs unnecssary files (libshared is statically linked into timew and does
not need to install any other files). Previously, the libshared source files
were listed explicitly so its install rules were not used.

Reported upstream in https://github.com/GothenburgBitFactory/timewarrior/issues/787

--- a/src/libshared/src/CMakeLists.txt
+++ b/src/libshared/src/CMakeLists.txt
@@ -56,10 +56,6 @@
 
 add_library (shared STATIC ${shared_SRCS})
 
-set (CMAKE_INSTALL_LIBDIR lib CACHE PATH "Output directory for libraries")
-install (TARGETS shared DESTINATION lib)
-install (FILES ${shared_HEADERS} DESTINATION include)
-
 add_executable (lex_executable lex.cpp)
 target_link_libraries (lex_executable shared)
 set_property (TARGET lex_executable PROPERTY OUTPUT_NAME "lex")
